### PR TITLE
Bpa enhancements

### PR DESCRIPTION
**Pull request by darkushin**
## Status
- [x] In Progress
- [ ] Ready
- [ ] In Hold - (Reason for hold)

## Related Issues
fixes: https://github.com/demisto/etc/issues/31950
fixes: https://github.com/demisto/etc/issues/31618
fixes: https://github.com/demisto/etc/issues/31343

## Description
Added new enhancements that allow to filter the results returned in `!pan-os-get-documentation` and `!pan-os-bpa-get-job-results` commands

## Must have
- [x] Tests
- [x] Documentation

### CODE REVIEW

File: Packs/BPA/Integrations/BPA/BPA.py [hunk 1]
```diff
@@ -120,21 +120,29 @@ def get_download_results_request(self, task_id: str) -> bytes:
...
-    filtered_docs = [filter_doc(doc) for doc in raw]
-
-    entry_context = {'PAN-OS-BPA.Documentation': filtered_docs}
-    human_readable = tableToMarkdown('BPA documentation', filtered_docs)
+    filter_by_ids = args.get('ids')
+    if filter_by_ids:
+        output = []
+        id_list = filter_by_ids.split(',')
+        for doc in raw:
+            if str(doc.get('doc_id')) in id_list:
```
Comment (yuvalbenshalom) [inline Packs/BPA/Integrations/BPA/BPA.py:139]: we can't filter this in the query?

File: Packs/BPA/Integrations/BPA/BPA.yml [hunk 2]
```diff
@@ -59,14 +114,30 @@ script:
...
       predefined:
-      - "true"
-      - "false"
-      defaultValue: "false"
+      - 'true'
+      - 'false'
+      required: false
+      secret: false
+    - default: false
+      description: A comma-separated list of the ids of the results to return.
```
Comment (yuvalbenshalom) [inline Packs/BPA/Integrations/BPA/BPA.yml:130]: what kind of ids?

File: Packs/BPA/Integrations/BPA/BPA.yml [hunk 3]
```diff
@@ -23,31 +23,86 @@ configuration:
...
+      required: false
+      secret: false
+    deprecated: false
     description: Gets the documentation of all BPA checks.
     execution: false
     name: pan-os-get-documentation
     outputs:
-    - contextPath: PAN-OS-BPA.Documentation
-      description: The list of BPA checks.
+    - contextPath: PAN-OS-BPA.Documentation.Document.DocId
```
Comment (yuvalbenshalom) [inline Packs/BPA/Integrations/BPA/BPA.yml:44]: I did not see the design of the outputs in the issues.

**Review by yuvalbenshalom — COMMENTED** [review]
looks good, added some questions

**Comment by ShirleyDenkberg** [comment]
@yuvalbenshalom Doc review completed.

**Review by yuvalbenshalom — APPROVED** [review]

**Pull request merged**